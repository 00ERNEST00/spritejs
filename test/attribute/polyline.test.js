import PolylineAttr from '../../src/attribute/polyline';

describe('attribute set get', () => {
  const subject = {
    onPropertyChange(key, value, oldValue, attributes) { },
  };
  const attr = new PolylineAttr(subject);

  test('set points', () => {
    expect(attr.d).toBe('');
    attr.points = [10, 10, 20, 20, 20, 10];
    expect(attr.d).toBe('M10 10L20 20L20 10');
    expect(attr.close).toBeFalsy();
    expect(attr.closeType).toBe('none');
    attr.close = true;
    expect(attr.close).toBeTruthy();
    expect(attr.d).toBe('M10 10L20 20L20 10Z');
    expect(attr.closeType).toBe('normal');
    expect(attr.smooth).toBeFalsy();
    attr.smooth = true;
    expect(attr.d).toMatchSnapshot();
    attr.points = [[10, 10], [20, 20], [10, 20]];
    attr.smooth = false;
    expect(attr.d).toBe('M10 10L20 20L10 20Z');
    attr.points = null;
    expect(attr.d).toBe('');
  });
});