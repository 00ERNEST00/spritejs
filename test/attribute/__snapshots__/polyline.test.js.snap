// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`attribute set get set points 1`] = `"M10 10C11.68 11.68 18.32 20 20 20C20 20 20 10 20 10Z"`;
